src/analysis/retail/priest/discipline/modules/spells/ShadowCovenant/ShadowCovSourceDonut.tsx

```diff
@@ -0,0 +1,65 @@
+import { formatThousands } from 'common/format';
+import { Ability } from 'parser/core/Events';
+import DonutChart from 'parser/ui/DonutChart';
+
+interface ScovSourceDonutProps {
+  abilityMap: Map<number, Ability>;
+  healingMap: Map<number, number>;
+}
+
+interface ScovSourceGraphItem {
+  color: string;
+  spellId: number;
+  label: string;
+  value: number;
+  valueTooltip?: string | number;
+}
+
+/**
+ * I'm not writing color values for every possible spell here
+ * https://gist.github.com/bendc/76c48ce53299e6078a76
+ */
+const randomInt = (min: number, max: number) => Math.floor(Math.random() * (max - min + 1)) + min;
+
+const randomColor = () => {
+  const h = randomInt(0, 360);
+  const s = randomInt(42, 98);
+  const l = randomInt(40, 90);
+  return `hsl(${h},${s}%,${l}%)`;
+};
```

Comment:
instead of using a random color, maybe use a color list. there may actually be one built in if you omit the `color` field :thinking: 

ideally, the color for a spell should always be the same. i don't know that we have good tools for that, but you might put some colors in for very common spells and then fall back to random or a color list for other spells.